### DISCUSSION TITLE
Change epochs and batch size

### DISCUSSION THREAD

**Discussion by mounismoumou** [Project-MONAI/MONAILabel#882] [Q&A] [answered]
Hello,

Sorry to bother you with this question, that seems pretty simple, but yet I can't find how to do it.

I want to change the number of epochs and batch size in the MONAI Label radiology app (not in 3D slicer options but in the scripts).

Thank you in advance for the time and the answer.

Mounir lahlouh.

**Comment by mounismoumou** [comment, marked as answer]
Finally, I found the answer to that thanks.
For those who couldn't change the trainning params, you go to the model configs script (for me it was deepedit.py in configs) and change the trainer function : 
config={"pretrained": strtobool(self.conf.get("use_pretrained_model", "true")), "max_epochs":100}

**Comment by SpirosMED** [comment]
Hello, could I ask how to use also tensorboard for MONAI?

Best,
Spiros

**Reply by diazandr3s** [reply]
Hi @SpirosMED,

If you use MONAI Label, by default you're using these Tensorboard handlers: https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/tasks/train/basic_train.py#L298

When you trigger training, MONAI Label automatically generates Tensorboard files that you can explore.

You could always add the handlers you wish by overwriting or augmenting the method: https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/tasks/train/basic_train.py#L286

Like this: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/trainers/segmentation.py#L151

Hope this helps,

**Reply by SpirosMED** [reply]
Hello, thank you for your answer, and is a great pleasure to finally talking to you . Seen all your MONAI Tutorials and they are very helpful. I will go one by one in your answer and get back to you because probably I will have some issues.

Hope you do not mind,

Best
Spiros

PS
My task is to segment the adipose tissue out of 3D MRI Volumes using MONAI. So far the prediction gives decent output but I want to optimise the parameters better.

**Reply by SpirosMED** [reply]
Hello again @diazandr3s 

I broke down your message as I said. However, I am having some issues since I am kind of new user of MONAI and know the basics. For your information I have opened a new discussion in a 3D Slicer Forum ( https://discourse.slicer.org/t/monai-pytorch-tensorboard-visualization/30120/2 ) where a couple of people try to help me out. 

As you can see from the forum, what I ask is simple enough : I want to monitor my training to optimise the number of epochs which I can use in order to avoid overfitting. In short the image below : 
![question_monai](https://github.com/Project-MONAI/MONAILabel/assets/137073228/521daf73-05af-4c42-9b4d-6a5335d46f82)

Since I am using 3D slicer with MONAI I am aware about the developer mode from which I can change the HyperParameters but in this case I am processing the data in a blind manner without getting a full picture.

--------------------------------------------------------------------------------------------------------------------------------------------------------

Regarding your previous answer : 

I am using the radiology app in a docker . Thus for my purpose I run the MONAI by : 

monailabel start_server --app /workspace/radiology --studies /workspace/Training/fat --conf models segmentationfat --conf multi_gpu true

Thus, where exactly is the .py script you mentioned in your answer ,named as basic_train .py  ( https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/tasks/train/basic_train.py#L298 ). 
I searched all the files of the model I am using but I cannot find it anywhere.


Secondly, the Tensorboardfiles are located in a path similar like this : 

''/radiology/model/segmentationfat/train_01/'' ?


Thirdly, how can I add the handlers to the segmentation.py file you mentioned ? I found this file in my case, named as segmentationfat.py and is exactly the same like the one you have in the link. What changed then?

Sorry in advance for the long answer, and maybe for my rookie questions

**Reply by diazandr3s** [reply]
Hi @SpirosMED,

Thanks for your questions.

> Thus, where exactly is the .py script you mentioned in your answer ,named as basic_train .py ( https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/tasks/train/basic_train.py#L298 ).
> I searched all the files of the model I am using but I cannot find it anywhere.

Are you working with the GitHub repository or have you installed monailabel via pip? In order to modify the basic_train.py file, you have to use the GitHub repo. The file is at   **MONAILabel/monailabel/tasks/train/basic_train.py** 

> Secondly, the Tensorboardfiles are located in a path similar like this :
> ''/radiology/model/segmentationfat/train_01/'' ?

That's correct

> Thirdly, how can I add the handlers to the segmentation.py file you mentioned ? I found this file in my case, named as segmentationfat.py and is exactly the same like the one you have in the link. What changed then?

Have you created a new Tensorboard handler? Just checked how a new Tensorboard handler is being added here: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/lib/trainers/segmentation.py#L151-L163
You can see this method is being extended.

Using the default handlers, you should be able to see losses and dice scores for training and validation. Not sure why you need more handlers. How many images are you using to train and validate the model? Is this a binary segmentation task, right?

Maybe you can explore this tool to check which transforms are best for your task: https://github.com/Project-MONAI/SlicerMONAIViz

Hope this helps,
